packages/session/session.0.3.1/opam

```diff
@@ -0,0 +1,57 @@
+opam-version: "1.2"
+maintainer: "Spiros Eliopoulos <spiros@inhabitedtype.com>"
+authors: [ "Spiros Eliopoulos <spiros@inhabitedtype.com>" ]
+license: "BSD-3-clause"
+homepage: "https://github.com/inhabitedtype/ocaml-session"
+dev-repo: "https://github.com/inhabitedtype/ocaml-session.git"
+bug-reports: "https://github.com/inhabitedtype/ocaml-session/issues"
+build: [
+  ["ocaml" "setup.ml" "-configure" "--prefix" prefix
+    "--%{webmachine:enable}%-webmachine"
+    "--%{postgresql:enable}%-postgresql"
+    "--%{redis:enable}%-redis"
+    "--%{lwt:enable}%-lwt"
+    "--%{cohttp:enable}%-cohttp"
+    "--%{async:enable}%-async"]
+  ["ocaml" "setup.ml" "-build"]
+]
+install: ["ocaml" "setup.ml" "-install"]
+remove: [
+  ["ocamlfind" "remove" "session"]
+]
+build-test: [
+  ["ocaml" "setup.ml" "-configure" "--enable-tests"
+    "--%{webmachine:enable}%-webmachine"
+    "--%{postgresql:enable}%-postgresql"
+    "--%{redis:enable}%-redis"
+    "--%{lwt:enable}%-lwt"
+    "--%{cohttp:enable}%-cohttp"
+    "--%{async:enable}%-async"]
+  ["ocaml" "setup.ml" "-build"]
+  ["ocaml" "setup.ml" "-test"]
+]
+build-doc: [ "ocaml" "setup.ml" "-doc" ]
+depends: [
+  "nocrypto"
+  "ocamlfind" {build}
+  "ounit" {test & >= "1.0.2"}
+  "result"
+]
+depopts: [
+  "async"
+  "base-threads"
+  "cohttp"
+  "postgresql"
+  # If session ever switches to jbuilder, then this constraint should turn into
+  # something along the lines of the following constraint in
+  # session-redis-lwt.opam:
+  #
+  #   depends: [
+  #     ...
+  #     ("redis-lwt" | ("redis" & "lwt"))
+  #     ...
+  #   ]
+  "lwt"
+  "redis-lwt"
```

Comment:
This doesn;'t quite work -- if lwt and redis are installed, it is activated in the build rule but `redis-lwt` may not be installed since it is a depopt. It would probably be easier to have a `session-redis` package.

```
#=== ERROR while installing session.0.3.1 =====================================#
# opam-version         1.2.2 (58ef3b8213100953848d362f7120a30356d7f77d)
# os                   linux
# command              ocaml setup.ml -configure --prefix /home/opam/.opam/4.03.0 --enable-webmachine --disable-postgresql --enable-redis --enable-lwt --enable-cohttp --disable-async
# path                 /home/opam/.opam/4.03.0/build/session.0.3.1
# compiler             4.03.0
# exit-code            1
# env-file             /home/opam/.opam/4.03.0/build/session.0.3.1/session-29-472e3a.env
# stdout-file          /home/opam/.opam/4.03.0/build/session.0.3.1/session-29-472e3a.out
# stderr-file          /home/opam/.opam/4.03.0/build/session.0.3.1/session-29-472e3a.err
### stderr ###
# ocamlfind: Package `redis-lwt' not found
# W: Field 'pkg_redis_lwt' is not set: Command ''/home/opam/.opam/4.03.0/bin/ocamlfind' query -format %d redis-lwt > '/tmp/oasis-cad565.txt'' terminated with error code 2
# E: Cannot find findlib package redis-lwt
# E: Failure("1 configuration error")

```